### PR TITLE
Fix multi rooted path tree scanning in the Qute processor

### LINKED ISSUE #40366: Qute-Templates cannot be found in tests if directory "src/test/resources/templates" exist
### Describe the bug

As soon as there is a directory `src/test/resources/templates`, Qute Templates can no longer be found when executing tests, resulting in a `No template matching the path foo/bar could be found for: org.acme.FooTemplates.bar`.

For more discussion, see https://github.com/quarkusio/quarkus/pull/40158

### Expected behavior

The existence of a `src/test/resources/templates` should not break the resolution of the templates.

### Actual behavior

`No template matching the path foo/bar could be found for: org.acme.FooTemplates.bar` when running tests.

### How to Reproduce?

Reproducer project: https://github.com/mzellho/quarkus-quickstarts/tree/reproducer/40158
- Run `mvn test` on `qute-quickstart` --> fail
- Delete `src/test/resources/templates` --> success

### Output of `uname -a` or `ver`

Linux xxxxx 6.5.0-28-generic #29-Ubuntu SMP PREEMPT_DYNAMIC Thu Mar 28 23:46:48 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "21.0.3" 2024-04-16 LTS OpenJDK Runtime Environment Temurin-21.0.3+9 (build 21.0.3+9-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (build 21.0.3+9-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.9.5

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Running `/home/xxxxx/work/2/quarkus-quickstarts/mvnw`...
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /home/xxxxx/.m2/wrapper/dists/apache-maven-3.8.6-bin/67568434/apache-maven-3.8.6
Java version: 21.0.3, vendor: Eclipse Adoptium, runtime: /home/xxxxx/.sdkman/candidates/java/21.0.3-tem
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "6.5.0-28-generic", arch: "amd64", family: "unix"

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by aloubyansky**
Fix https://github.com/quarkusio/quarkus/issues/40366 and https://github.com/quarkiverse/quarkus-web-bundler/issues/197

### CODE REVIEW

**Comment by aloubyansky** [comment]
Given https://github.com/quarkusio/quarkus/pull/40158 has `backport-3.8` label, i added it to this PR as well

File: independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathVisit.java [hunk 1]
```diff
@@ -44,6 +44,16 @@ default URL getUrl() {
...
     }
 
+    /**
+     * Path relative to the root of the tree as a string with {@code /} as a path element separator.
+     * This method calls {@link #getRelativePath(String)} passing {@code /} as an argument.
+     *
+     * @return path relative to the root of the tree as a string with {@code /} as a path element separator
+     */
+    default String getRelativePath() {
+        return getRelativePath("/");
```
Comment (ia3andy) [inline independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathVisit.java:54]: @aloubyansky  is this also working when fs uses "\" ?

Comment (aloubyansky) [reply independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathVisit.java:54]: Yes, it's just a string with '/' as a separator.

**Review by ia3andy — APPROVED** [review]
Looking good!

So if I understand well:
You use the `getRelativePath` which is directly using the given path separator, so no os problems. Also the relativise doesn't need to be done using the full Path, it's just stripping a part of the string, therefore no fs issues.

**Comment by mkouba** [comment]
Looks good but I wanted to add a test with `src/main/resources/templates` and `src/test/resources/templates` template roots. But never mind, I'll add one in a follow-up pr.

**Pull request merged**